src/main/java/tools/jackson/databind/cfg/EnumFeature.java

```diff
@@ -1,27 +1,97 @@
      * index of <code>Enum</code>;
      *<p>
      * Note: this feature should be symmetric to
-     * as {@link SerializationFeature#WRITE_ENUM_KEYS_USING_INDEX}.
+     * as {@link #WRITE_ENUM_KEYS_USING_INDEX}.
      *<p>
      * Feature is disabled by default.
      *
      * @since 2.15
      */
     READ_ENUM_KEYS_USING_INDEX(false),
 
+
+    /**
+     * Feature that determines the deserialization mechanism used for
+     * Enum values: if enabled, Enums are assumed to have been serialized using
+     * return value of {@code Enum.toString()};
+     * if disabled, return value of {@code Enum.name()} is assumed to have been used.
+     *<p>
+     * Note: this feature should usually have same value
+     * as {@link #WRITE_ENUMS_USING_TO_STRING}.
+     *<p>
+     * Feature is enabled by default as of Jackson 3.0 (in 2.x it was disabled).
+     */
+    READ_ENUMS_USING_TO_STRING(true),
+
+    /**
+     * Feature that allows unknown Enum values to be parsed as {@code null} values.
+     * If disabled, unknown Enum values will throw exceptions.
+     * <p>
+     * Note that in some cases this will effectively ignore unknown {@code Enum} values,
+     * e.g. when the unknown values are used as keys of {@link java.util.EnumMap}
+     * or values of {@link java.util.EnumSet}: this is because these data structures cannot
+     * store {@code null} values.
+     * <p>
+     * Also note that this feature has lower precedence than
+     * {@link #READ_UNKNOWN_ENUM_VALUES_USING_DEFAULT_VALUE},
+     * meaning this feature will work only if latter feature is disabled.
+     * <p>
+     * Feature is disabled by default.
+     */
+    READ_UNKNOWN_ENUM_VALUES_AS_NULL(false),
+
+    /**
+     * Feature that allows unknown Enum values to be ignored and replaced by a predefined value specified through
+     * {@link com.fasterxml.jackson.annotation.JsonEnumDefaultValue @JsonEnumDefaultValue} annotation.
+     * If disabled, unknown Enum values will throw exceptions.
+     * If enabled, but no predefined default Enum value is specified, an exception will be thrown as well.
+     * <p>
+     * Note that this feature has higher precedence than {@link #READ_UNKNOWN_ENUM_VALUES_AS_NULL}.
+     * <p>
+     * Feature is disabled by default.
+     */
+    READ_UNKNOWN_ENUM_VALUES_USING_DEFAULT_VALUE(false),
+
+    /**
+     * Feature that determines whether JSON integer numbers are valid
+     * values to be used for deserializing Java enum values.
+     * If set to 'false' numbers are acceptable and are used to map to
+     * ordinal() of matching enumeration value; if 'true', numbers are
+     * not allowed and a {@link DatabindException} will be thrown.
+     * Latter behavior makes sense if there is concern that accidental
+     * mapping from integer values to enums might happen (and when enums
+     * are always serialized as JSON Strings)
+     *<p>
+     * Feature is disabled by default.
+     */
+    FAIL_ON_NUMBERS_FOR_ENUMS(false),
```

Comment:
Let's try to keep alphabetical within sections, so this should go before first `READ_` feature?